File: app/src/main/java/org/dhis2/utils/granular_sync/SyncConflictHolder.java
package org.dhis2.utils.granular_sync;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import org.dhis2.databinding.ItemSyncConflictBinding;
import org.dhis2.utils.DateUtils;

class SyncConflictHolder extends RecyclerView.ViewHolder {
    private final ItemSyncConflictBinding binding;

    SyncConflictHolder(@NonNull ItemSyncConflictBinding binding) {
        super(binding.getRoot());
        this.binding = binding;
    }

    public void bind(StatusLogItem trackerImportConflict) {
        binding.date.setText(DateUtils.dateTimeFormat().format(trackerImportConflict.date()));
        binding.message.setText(trackerImportConflict.description());
    }
}

File: app/src/main/java/org/dhis2/utils/granularsync/SyncConflictHolder.kt
package org.dhis2.utils.granularsync

import androidx.recyclerview.widget.RecyclerView
import org.dhis2.commons.date.DateUtils
import org.dhis2.databinding.ItemSyncConflictBinding

class SyncConflictHolder(
    private val binding: ItemSyncConflictBinding,
) : RecyclerView.ViewHolder(binding.root) {
    fun bind(trackerImportConflict: StatusLogItem) {
        binding.date.text = DateUtils.dateTimeFormat().format(trackerImportConflict.date)
        binding.message.text = trackerImportConflict.description
    }
}
